### PR TITLE
Share configuration between Rebar3 and Mix

### PR DESCRIPTION
**Pull request by hauleth**
This should reduce amount of places where the configuration mismatch can
happen. Current solution uses Rebar3 configuration as a source of truth.

TBD:

- support for Rebar3 profile sharing

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -17,24 +21,49 @@ defmodule OpenTelemetry.MixProject do
...
+  end
+
+  defp load_config do
+    {:ok, config} = :file.consult('rebar.config')
+
+    config
+  end
+
+  defp load_app do
+    {:ok, [{:application, name, desc}]} = :file.consult('src/opentelemetry_api.app.src')
```
Comment (tsloughter) [inline mix.exs:85]: This wouldn't work when used as a dependency if it was rebar3 running this script, but I take it `mix` changes directory to the root of each dep before building it? Did you verify this works if used as a git dep of another mix project?

**Review by tsloughter — APPROVED** [review]
Just had the one question about reading in the `.app.src` file from a relative path.

File: mix.exs (hunk 1 above)
Comment (zachdaniel) [reply mix.exs:85]: I'm pretty sure it would work, but if not then `File.cwd()` will always be the project root.

**Review by zachdaniel — APPROVED** [review]
Should double check using as a git dependency just to be sure.

File: mix.exs (hunk 1 above)
Comment (tsloughter) [reply mix.exs:85]: What does "project root" mean in this context? If by project you mean the opentelemetry_api application, then good and there should be no need to add the path. 

But if "project root' means the project that has opentelemetry_api as a dependency then the current code won't work and not sure what you are to do with the cwd if that is the case.

Comment (zachdaniel) [reply mix.exs:85]: Sorry, that was unclear. I really just meant "it should be what you think it is", as in the root of *this*(opentelementry-api) project. I actually just confirmed this using some of my own repos.

**Comment by tsloughter** [comment]
@hauleth can you push the updated mix.lock file to this?

**Review by tsloughter — APPROVED** [review]

**Pull request merged**